File: src/main/java/lana/thingService/thing/Thing.java
package lana.thingService.thing;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import lana.thingService.attribute.Attribute;
import lana.thingService.attribute.binding.AttributeJsonDeserializer;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;

@Getter
@Setter
@NoArgsConstructor
@Entity
public class Thing {
    @Id
    @GeneratedValue
    private Integer id;

    private String name;

    private String description;

    //TODO: make new model Generic
    private String generic;

    @ManyToOne
    @JoinColumn
    // to binding attribute from id.
    @JsonDeserialize(using = AttributeJsonDeserializer.class)
    private Attribute attribute;
}

File: src/main/java/lana/thingService/thing/Thing.kt
package lana.thingService.thing;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize
import lana.thingService.attribute.Attribute
import lana.thingService.attribute.binding.AttributeJsonDeserializer

import javax.persistence.*

@Entity
@Table(name = "thing")
class Thing(
        @Id
        @GeneratedValue
        var id: Int? = null,

        var name: String = "",

        var description: String = "",

        var generic: String = "",

        @ManyToOne
        @JoinColumn
        @JsonDeserialize(using = AttributeJsonDeserializer::class)
        var attribute: Attribute? = null
)